_client/src/Views/MainWindow.cpp

```diff
@@ -25,6 +28,47 @@ void MainWindow::setupUI()
     this->showMaximized();
 }
 
+void MainWindow::setupMenu() 
+{
+    // create MenuBar
+    m_mainMenuBar =  menuBar();
+
+    // File Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("File"), {ActionTypes::SETTINGS, ActionTypes::EXIT}));
+    
+    // Purchasing Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Purchasing"), {ActionTypes::PURCHASE_ORDERS, ActionTypes::SUPPLIER_MANAGEMENT, ActionTypes::GOODS_RECEIPTS}));
+
+    // Sales Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Sales"), {ActionTypes::SALES_ORDERS, ActionTypes::CUSTOMER_MANAGEMENT, ActionTypes::INVOICING}));
+    
+    // Inventory Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Inventory"), {ActionTypes::PRODUCT_MANAGEMENT, ActionTypes::SUPPLIER_PRICELIST_UPLOAD, ActionTypes::STOCK_TRACKING}));
+
+    // Analytics Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Analytics"), {ActionTypes::SALES_ANALYTICS, ActionTypes::INVENTORY_ANALYTICS}));
+
+    // Users Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Users"), {ActionTypes::USER_ROLES, ActionTypes::USER_LOGS}));
+
+    // Management Module Menu
+    m_mainMenuBar->addMenu(createModuleMenu(tr("Management"), {ActionTypes::EMPLOYEES, ActionTypes::CUSTOMERS, ActionTypes::SUPPLIERS}));
+}
+
+//Custom function that create module menu using map
+QMenu* MainWindow::createModuleMenu(const QString& menuTitle, const std::vector<ActionTypes>& actions)
+{
+    QMenu* menu = new QMenu(menuTitle);
+
+    for (ActionTypes actionType : actions) {
+        QAction* action = new QAction(ACTION_NAMES.at(actionType).c_str(), this);
```

Comment:
Let menu, that you create on line 61 manage lifetime of this actions. Isn't it logical?
Make menu a parent of it's actions

Reply:
done